pandas/core/frame.py

```diff
@@ -3604,7 +3609,11 @@ def transpose(self, *args, copy: bool = False) -> DataFrame:
             if copy:
                 new_arr = new_arr.copy()
```

Comment:
If we can pass copy=False below (which I think is correct, since in this code path we know that we have multiple blocks, and so `self.values` will always be a copy), then this copy if `copy=True` should also not be necessary? 

I would also add a short comment mentioning why we can specify `copy=False`

Reply:
Yeah I thought so as well, couldn't figure out why we would do an additional copy here, but wasn't sure if I am missing something.

Added a comment